### ISSUE TITLE
Synchronization is too slow and only few nodes keep connection.

### ISSUE THREAD

**Issue by gogumia** [gxchain/gxb-core#109]
**Describe the bug**
the title is the problem what I have.
Should I set another configurations like seed-node? 

Another question. Which one is most correct technical document?
1. https://doc.gxb.io/gxchain/
2. https://github.com/gxchain/gxb-core/wiki
3. https://github.com/gxchain/docs

**To Reproduce**
1. run witness_node and start synchronize(first run)

$ ./bin/witness_node --data-dir=./witness_node_data_dir --rpc-endpoint=127.0.0.1:28090 --p2p-endpoint=0.0.0.0:6789 --log-file --track-account "\"1.2.1004580\"" --partial-operations=true --max-ops-per-account=100 --plugins account_history

2. I checked the correct log in witness.log

 tail -f witness_node_data_dir/logs/witness.log
1233382ms th_a       application.cpp:497           handle_block         ] Got block: #9850000 time: 2018-05-28T01:46:15 transaction(s): 8 latency: 18963258382 ms from: miner8  irreversible: 9849980 (-20)
1574439ms th_a       application.cpp:497           handle_block         ] Got block: #9860000 time: 2018-05-28T10:13:54 transaction(s): 19 latency: 18933140439 ms from: aaron  irreversible: 9859985 (-15)
1908886ms th_a       application.cpp:497           handle_block         ] Got block: #9870000 time: 2018-05-28T18:41:33 transaction(s): 4 latency: 18903015886 ms from: kairos  irreversible: 9869985 (-15)
2218693ms th_a       application.cpp:497           handle_block         ] Got block: #9880000 time: 2018-05-29T03:09:03 transaction(s): 5 latency: 18872875693 ms from: miner5  irreversible: 9879982 (-18)
2534839ms th_a       application.cpp:497           handle_block         ] Got block: #9890000 time: 2018-05-29T11:36:42 transaction(s): 13 latency: 18842732838 ms from: sakura  irreversible: 9889982 (-18)
2850859ms th_a       application.cpp:497           handle_block         ] Got block: #9900000 time: 2018-05-29T20:04:21 transaction(s): 1 latency: 18812589859 ms from: david12  irreversible: 9899980 (-20)
3163127ms th_a       application.cpp:497           handle_block         ] Got block: #9910000 time: 2018-05-30T04:32:00 transaction(s): 9 latency: 18782443127 ms from: miner11  irreversible: 9909985 (-15)
....

3. But synchronization is too slow.. It's been over 6 hours since I started witness_node,
also I check connection status by using lsof 
$ lsof -i -nP 
witness_n 9239  gxs   10u  IPv4  59609      0t0  TCP *:6789 (LISTEN)
witness_n 9239  gxs   12u  IPv4 134725      0t0  TCP 10.0.2.15:6789->47.104.0.35:6789 (ESTABLISHED)
witness_n 9239  gxs   16u  IPv4 134728      0t0  TCP 10.0.2.15:6789->47.74.22.124:6789 (ESTABLISHED)
witness_n 9239  gxs   19u  IPv4  44561      0t0  TCP 127.0.0.1:28090 (LISTEN)
witness_n 9239  gxs   22u  IPv4 134767      0t0  TCP 10.0.2.15:6789->47.75.71.70:6789 (CLOSE_WAIT)
witness_n 9239  gxs   23u  IPv4 134768      0t0  TCP 10.0.2.15:6789->178.128.59.227:33603 (CLOSE_WAIT)
witness_n 9239  gxs   24u  IPv4 134769      0t0  TCP 10.0.2.15:6789->47.75.148.12:6789 (CLOSE_WAIT)
witness_n 9239  gxs   25u  IPv4 134770      0t0  TCP 10.0.2.15:6789->47.254.146.219:6789 (CLOSE_WAIT)

Only few node established and another nodes disconnect as soon as it's connected.

4. I checked P2P.log
 
It keeps print out following error message

$ tail -f witness_node_data_dir/logs/p2p/p2p.log
2019-01-02T14:26:09 p2p:message read_loop            read_loop ] disconnected 0 exception: unspecified
Bad file descriptor
    {"message":"Bad file descriptor"}
    asio  asio.cpp:41 operator()

    {"len":16}
    p2p  stcp_socket.cpp:117 readsome                   message_oriented_connection.cpp:205

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
    CentOS 7 1804
    SSD 200GB
    32GB RAM
    4 cores
    Giga speed internet

$ ./bin/witness_node --version
Version: v1.0.181212
SHA: 7b6209f0171d0eb4ef8395d332396590a6ec24a9
Timestamp: 21 days ago
SSL: OpenSSL 1.0.2p  14 Aug 2018
Boost: 1.67
Websocket++: 0.7.0

**Smartphone (please complete the following information):**

**Additional context**

**Your GXChain Wallet Account**
Please leave your GXChain wallet account for bug bounty .

**Comment by zhuliting** [comment]
@gogumia 
Current there are about 16,000,000+ blocks, and it will take about 30 hours syncing all the blocks from gxchain network.
The latest document is https://github.com/gxchain/docs or https://docs.gxchain.org/

**Comment by gogumia** [comment]
@zhuliting 
Thanks for your quick support. 
Then, I'll have to wait patiently. Okey..

**Comment by gogumia** [comment]
@zhuliting 

I think all node should update to gxb-core181226... 
As I mentioned here, 
```
3. But synchronization is too slow.. It's been over 6 hours since I started witness_node,
also I check connection status by using lsof
$ lsof -i -nP
witness_n 9239 gxs 10u IPv4 59609 0t0 TCP *:6789 (LISTEN)
witness_n 9239 gxs 12u IPv4 134725 0t0 TCP 10.0.2.15:6789->47.104.0.35:6789 (ESTABLISHED)
witness_n 9239 gxs 16u IPv4 134728 0t0 TCP 10.0.2.15:6789->47.74.22.124:6789 (ESTABLISHED)
witness_n 9239 gxs 19u IPv4 44561 0t0 TCP 127.0.0.1:28090 (LISTEN)
witness_n 9239 gxs 22u IPv4 134767 0t0 TCP 10.0.2.15:6789->47.75.71.70:6789 (CLOSE_WAIT)
witness_n 9239 gxs 23u IPv4 134768 0t0 TCP 10.0.2.15:6789->178.128.59.227:33603 (CLOSE_WAIT)
witness_n 9239 gxs 24u IPv4 134769 0t0 TCP 10.0.2.15:6789->47.75.148.12:6789 (CLOSE_WAIT)
witness_n 9239 gxs 25u IPv4 134770 0t0 TCP 10.0.2.15:6789->47.254.146.219:6789 (CLOSE_WAIT)
```
I ran two nods gxb-core-181212 and gxb-core-181226 separately, gxb-core-181212 very few nodes(1~2) maintain connection and in case of gxb-core-181226 very many nodes keep connection.

**Comment by zhuliting** [comment]
There are no P2P module related changes between the two versions.

**Comment by gogumia** [comment]
@zhuliting 

Yes, I know but It seemed like nodes disconnect my connection. 

If node connection is unstable, block chain data is broken very easily. :(

I ran 5 different nodes separately and tried synchronization since 181212 version released. 
now Only 1 node has version 181226 successfully synced and maintain most recent height.
The others block chain data corruption and resynchronization are being repeated over two weeks.

Yesterday I upgraded all nodes to version 181226 and is waiting for the re-sync to finish.
Let's see the result.
